### ISSUE TITLE
"searchall" parameter behavior in events/restSearch and attributes/restSearch

### ISSUE THREAD

**Issue by obert01** [MISP/MISP#4205]
Hello,

When I use the "searchall" parameter in a restSearch query, it appears that all the other parameters of the requests are ignored. I know this behavior is intended with "quickFilter":1, but I expected "searchall" to honor all the other parameters. 

Let's say for instance I want to obtain all md5 hashes for events related to WannaCry. Here is my request:

```
curl -s -o - -X POST --header "Authorization: XXXX" \
  --header "Content-Type: application/json" \
  --data '{
         "returnFormat":"csv",
         "searchall":"%wannacry%",
         "to_ids":1,
         "enforceWarninglist":1,
         "type": ["md5"],
         "requested_attributes":["event_info","type","value","comment"]
}' \
https://misp.cert-ist.com/attributes/restSearch
```
This request seems to return all attributes belonging to events where the term "WannaCry" has been found, while I just want md5. 

An idea? Is it a bug? What is the difference between "searchall" and "quickFilter"?

**Comment by iglocska** [comment]
I will have to dig into this, but afaik searchall is just an alias for quickFilter due to some nasty misalignments between the documentation and reality (so they now do the same thing).

**Comment by obert01** [comment]
This confirms my experiments. But it is damage that you can't set such a filter without ignoring all other filters. 

Moreover, I note that most restSearch filters ("tags", "eventinfo", "value", ...) are case sensitive, which appears counter-intuitive to me.  For example, searching for the tag "trickbot" is not the same as searching for "TrickBot". I was about to prepare a pull request to fix this, but I don't know if it would be phylosophically OK for you.

**Comment by obert01** [comment]
Finally, I am also not able to restSearch for events/attributes containing several tags ("AND" operation). 

I have tried using a simple string value like "tag1&tag2", "tag1&&tag2", and finally some JSON constructions like { "AND" : ["tag1", "tag2"]} ... no success.

**Comment by iglocska** [comment]
Indeed it's a good point about not being able to do so being harmful. I'll look into some way of resolving that.

AND is not implemented only OR and NOT so far. On my todo list for sure, will try to do it soon, that request keeps haunting me :)

**Comment by obert01** [comment]
What about the fact that most restSearch filters are case sensitive? Would you agree on making the filter case insensitive for "searchall", "value", "eventinfo", and possibly "tags"?

**Comment by iglocska** [comment]
yes and no. Definitely something we should do over time, but we have a dependency that is blocking us for now. Basically we need to tune it on mysql level so that the tables are CI - but we currently have no way of running updates that are as slow as updating the attribute table. It's in progress though!

**Comment by ITServ-DE** [comment]
Any news here? 

Even though I do not have an issue with this right now, I suspect I will run in problems when interfacing with other tools, which might store hashes uppercase. And I'm not sure every feed provider uses lowercase hashes/tags/...

I don't see the chance to make the underlying mySQL DB case insensitive by itself (to be done during table creation), so I think this point should be done in MISP by running only case-normalized searches:

`SELECT .... FROM .... WHERE lower ({field}) = lower ({user entry})`

What do you think?

**Comment by pettai** [comment]
Ping? Has there been any progress with regards to "searchall" and type filtering ?
